src/instruments/src/EFB/Ground/Ground.tsx

```diff
@@ -17,47 +17,142 @@
+import { IconCornerDownLeft, IconCornerDownRight, IconArrowDown, IconHandStop, IconTruck, IconBriefcase, IconBuildingArch, IconArchive } from '@tabler/icons'
 
 import './Ground.scss'
 import fuselage from '../Assets/320neo-outline-upright.svg'
+import {setSimVar, getSimVar} from '../../util.mjs'
 
 type GroundProps = {}
 
-type GroundState = {}
+type GroundState = {
+    tugActive: boolean;
+    activeButton: string,
+}
 
 class Ground extends React.Component<GroundProps, GroundState> {
 
+    constructor(props: GroundProps) {
+        super(props);
+
+        this.state = {
+            tugActive: false,
+            activeButton: 'none',
+        };
+      }
+
+    toggleGroundAction(action: GroundServices) {
+            setSimVar(action, "1", "boolean");
+    }
+
+    setTugHeading(action: GroundServices, direction: number) {
+            if (!this.state.tugActive) {
+                this.togglePushback();
+            } else {
+                const tugHeading = this.getTugHeading(direction);
+                // KEY_TUG_HEADING is an unsigned integer, so let's convert
+                setSimVar(action, (tugHeading * 11930465) & 0xffffffff, "UINT32");
+            }
+    }
+
+    getTugHeading(value: number): number {
+        const currentHeading = getSimVar("PLANE HEADING DEGREES TRUE", "degrees");
+        return (currentHeading  + value) % 360;
+    }
+
+    togglePushback() {
+        const tugActive = this.state.tugActive;
+        setSimVar(GroundServices.TOGGLE_PUSHBACK, 1, "boolean");
+        this.setState({
+            tugActive: !tugActive
+        });
+    }
+
+    unselectButton() {
+        this.setState({
+            activeButton: "none"
+        })
+    }
+
+    handleClick(callBack: () => void, event: React.MouseEvent) {
+        this.setState({
+            activeButton: event.currentTarget.id
+        });
+        callBack();
+    }
+
     render() {
         return (
             <div className="wrapper flex-grow flex flex-col">
+                <img className="airplane w-full" src={fuselage} />
                 <div className="pushback control-grid">
                     <h1 className="text-white font-medium text-xl">Pushback</h1>
-                    <div className="stop"><IconHandStop/></div>
-                    <div className="down-left"><IconCornerDownLeft/></div>
-                    <div className="down selected"><IconArrowDown /></div>
-                    <div className="down-right"><IconCornerDownRight/></div>
+                    <div id="stop"
+                         onMouseDown={(e) => this.handleClick(() => this.state.tugActive ? this.togglePushback() : {}, e)}
+                         onMouseUp={() => this.unselectButton()}
+                         className={this.state.activeButton === "stop" ? "stop selected" : "stop"}><IconHandStop/>
```

Comment:
I saw in the discord that you were worried about how wordy the code for the selected style was. What about a helper function like this:

```
applySelected(className: string) {
   return className + (this.state.activeButton === className ? ' selected' : '');
}
```

and then here you could have:

```
className={this.applySelected('stop')}
```

Reply:
Heh that makes sense.